moveit_ros/planning_interface/test/subframes_test.cpp

```diff
@@ -176,15 +175,25 @@ TEST(TestPlanUsingSubframes, SubframesTests)
     planning_scene_monitor::LockedPlanningSceneRO planning_scene(planning_scene_monitor);
 
     // get the tip and box subframe locations in world
-    Eigen::Isometry3d eef = planning_scene->getFrameTransform("cylinder/tip");
+    // TODO (felixvd): Get these from the plan's goal state instead, so we don't have to execute the motion in CI
+    Eigen::Isometry3d cyl_tip = planning_scene->getFrameTransform("cylinder/tip");
     Eigen::Isometry3d box_subframe = planning_scene->getFrameTransform(target_pose_stamped.header.frame_id);
     Eigen::Isometry3d target_pose;
     tf2::fromMsg(target_pose_stamped.pose, target_pose);
 
     // expect that they are identical
     std::stringstream ss;
-    ss << "target frame: \n" << (box_subframe * target_pose).matrix() << "\ncylinder frame: \n" << eef.matrix();
-    EXPECT_TRUE(eef.isApprox(box_subframe * target_pose, EPSILON)) << ss.str();
+    ss << "target frame: \n" << (box_subframe * target_pose).matrix() << "\ncylinder frame: \n" << cyl_tip.matrix();
+    EXPECT_TRUE(cyl_tip.isApprox(box_subframe * target_pose, EPSILON)) << ss.str();
+
+    // Check that robot wrist is where we expect it to be
+    Eigen::Isometry3d panda_link = planning_scene->getFrameTransform("panda_link8");
+    Eigen::Isometry3d expected_pose = Eigen::Isometry3d(Eigen::Translation3d(0.307, 0.09, 0.44)) *
```

Comment:
This's why it's failing, I think the value should be `0.13`, not `0.09` I launch RViz and everything seem to work ok

```
Value of: panda_link.isApprox(expected_pose, EPSILON)
  Actual: false
Expected: true
panda link frame: 
    -0.70632    -0.707893 -0.000267668     0.306952
   -0.707893     0.706319  0.000416174     0.129875
-0.000105548  0.000483432           -1     0.440295
           0            0            0            1
expected pose: 
   -0.706554    -0.707633  0.000488458        0.307
   -0.707632     0.706576  0.000810873         0.09
-0.000918937  0.000227289    -0.999978         0.44
           0            0            0            1
```

Reply:
How strange. These tests passed before. Was there a change in the `panda_moveit_config`? I didn't manage to check this visually in the subframes tutorial, because the cylinder is allegedly in collision with the panda hand (although it clearly isn't). Does it work for you?

Reply:
> These tests passed before. Was there a change in the panda_moveit_config? 

We're not using `panda_moveit_config` anymore in the tests [see](https://github.com/ros-planning/moveit/pull/2300)

> I didn't manage to check this visually in the subframes tutorial, because the cylinder is allegedly in collision with the panda hand (although it clearly isn't). Does it work for you?

Yes, it works for me, and the robot moves without any problem

Reply:
I thought we changed `ros-planning/panda_moveit_config`. Actually that means that the subframes tutorial doesn't work anymore. Oh well, that's for a separate issue I suppose. 

The difference in the `expected_pose` was because the Z_OFFSET `subframes_test.cpp` was not the same as in `subframes_tutorial.cpp`. I wrongly assumed that it was, because the test is based on the tutorial.

Reply:
How do you run the test itself by the way? `rostest moveit_ros_planning_interface` doesn't tab-complete to anything for me.

Reply:
Make sure to build the tests first `catkin build  moveit_ros_planning_interface --no-deps --make-args tests`